packages/native-stack/src/views/NativeStackView.native.tsx

```diff
@@ -461,6 +467,12 @@ export function NativeStackView({ state, navigation, descriptors }: Props) {
             ? descriptors[previousKey]
             : undefined;
           const nextDescriptor = nextKey ? descriptors[nextKey] : undefined;
+          if (
+            !descriptor.options.presentation &&
+            previousDescriptor?.options.presentation === 'modal'
+          ) {
+            isForsedModalStack = true;
```

Comment:
Changing an outside variable and using it in a map may be unexpected when reading the code since the function passed to map is usually expected to be a pure function.

It might be better to do something like:

```js
const modalRouteKeys = state.routes.reduce((modals, route) => {
  const { presentation } = descriptors[route.key].options;

  // Assume modal if there are already modal screens in the stack
  // or current screen is a modal when no presentation is specified
  if ((modals.length && !presentation) || presentation === 'modal') {
    acc.push(route.key);
  }

  return acc;
}, []);
```

And later check if something is modal by `const isModal = modalRouteKeys.includes(route.key)`